src/Parser/Hunter/BeastMastery/Modules/Spells/BestialWrath/BestialWrathUptime.js

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS/index';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import StatisticBox from "Main/StatisticBox";
+import SpellIcon from "common/SpellIcon";
+import { formatPercentage, formatNumber } from "common/format";
+import Icon from "common/Icon";
+import SpellLink from "common/SpellLink";
+
+class BestialWrathUptime extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  bestialWrathCasts = 0;
+  accumulatedFocusAtBWCast = 0;
+
+  get percentUptime() {
+    //This calculates the uptime over the course of the encounter of Bestial Wrath
+    const uptime = this.combatants.selected.getBuffUptime(SPELLS.BESTIAL_WRATH.id) / this.owner.fightDuration;
+    return uptime;
+  }
+
+  on_byPlayer_cast(event) {
+    const buffId = event.ability.guid;
+    if (buffId !== SPELLS.BESTIAL_WRATH.id) {
+      return;
+    }
+    this.bestialWrathCasts += 1;
+    this.accumulatedFocusAtBWCast += event.classResources[0]['amount'] || 0;
+  }
+  statistic() {
+    const averageFocusAtBW = formatNumber(this.accumulatedFocusAtBWCast / this.bestialWrathCasts);
+    return (
+      <StatisticBox
+        icon={<SpellIcon id={SPELLS.BESTIAL_WRATH.id} />}
+        value={(
+          <span>
+          {formatPercentage(this.percentUptime)}{`%`}
+            <br/>
+            {averageFocusAtBW}{'  '}
+            <Icon
+              icon='ability_hunter_focusfire'
+              alt='Average Focus At Trueshot Cast'
+              style={{
+                height: '1.3em',
+                marginTop: '-.1em',
+              }}
+            />
+          </span>
+        )}
+        label={`Bestial Wrath information`}
+        tooltip={`You started your average Bestial Wrath window with ${averageFocusAtBW} focus, and you had an overall uptime of ${formatPercentage(this.percentUptime)}% on Bestial Wrath.`}
```

Comment:
Looks like you put two unrelated statistics in the same box, and it's causing display issues. As I suggested above, maybe drop the uptime display.

Reply:
I can split the two modules to showcase information better, but I think it's quite relevant to show uptime, as a large part of the spec revolves around reducing the cooldown of BestialWrath and optimising damage inside it, so having low uptime would suggest incorrect play.

Reply:
Well then isn't uptime just a poor proxy for cast efficiency in this case?

As for optimizing damage inside of Bestial Wrath, that does sound like a useful stats box.

Reply:
I need to do some work on Cast Efficiency, but it's a bit of a pita because the spec has 2 untracked cooldown reducing/resetting events that happen often on spells that reduce Bestial Wrath cooldown (so cast efficiency on BW is currently saying stuff like 7/3). 

I asked the BM experts, and they'd like an uptime module atleast as long as Cast Efficiency is bonkered. 

I'm not sure if damage inside BW windows is possible without creating 3 or 4 versions because of different viable builds, will look into it.